### ISSUE TITLE
shapeless.the does not respect @implicitNotFound

### ISSUE THREAD

**Issue by Atry** [milessabin/shapeless#671]
``` scala
@ Ordering[Set[Int]] 
cmd5.scala:1: No implicit Ordering defined for Set[Int].
val res5 = Ordering[Set[Int]]
                   ^
Compilation Failed
@ type StringOrderingOps = the.`Ordering[Set[Int]]`.Ops 
cmd5.scala:1: Could not infer implicit value for Ordering[Set[Int]]
type StringOrderingOps = the.`Ordering[Set[Int]]`.Ops
                         ^
Compilation Failed
```

There is an `@implicitNotFound` annotation on `Ordering`, thus I expect the error message for ```the.`Ordering[Set[Int]]`.Ops``` is `No implicit Ordering defined for Set[Int].`. However the error message is `Could not infer implicit value for Ordering[Set[Int]]`

**Comment by milessabin** [comment]
Good catch!

**Comment by Atry** [comment]
Fixed by #742
